.github/workflows/dependent-tests.yml

```diff
@@ -29,16 +29,13 @@ jobs:
     - name: Install dependencies
       run: |
         pip install --upgrade pip
-        pip install --upgrade git+https://github.com/aws/amazon-braket-schemas-python.git@main
-        pip install --upgrade git+https://github.com/aws/amazon-braket-default-simulator-python.git@main
-        pip install -e .
         cd ..
         git clone https://github.com/aws/${{ matrix.dependent }}.git
         cd ${{ matrix.dependent }}
         # Update the amazon-braket-sdk dependency to reference the current commit
         python ${GITHUB_WORKSPACE}/.github/scripts/update_dependency.py
-        pip install -e .[test]
+        pip install tox
```

Comment:
Could you validate if this causes the tests to reference the SDK mainline instead of the current commit?

Reply:
I am unsure of the behavior when you left this comment so I added a deterministic change to the tox file to ensure the setup uses the working branch for the PR.

From the test [logs](https://github.com/amazon-braket/amazon-braket-sdk-python/actions/runs/6897498200/job/18765787081?pr=792):

```
unit-tests: install_deps> python -I -m pip install git+https://github.com/amazon-braket/amazon-braket-sdk-python.git@math411-patch-4
```